### PR TITLE
Fix enum bug

### PR DESCRIPTION
**Pull request by yangdsh**

### CODE REVIEW

File: server/website/website/db/base/parser.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def create_knob_configuration(self, tuning_knobs):
         return configuration
 
     def format_bool(self, bool_value, metadata):
-        return self.true_value if bool_value == BooleanType.TRUE else self.false_value
+        return self.true_value if int(round(bool_value)) == BooleanType.TRUE else self.false_value
```
Comment (yangdsh) [inline server/website/website/db/base/parser.py]: I think we need to convert the recommended value to an integer.

File: server/website/website/management/commands/resetwebsite.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def __init__(self, *args, **kwargs):
         port = DATABASES['default']['PORT']
 
         if engine.endswith('mysql'):
-            db_cmd_fmt = 'mysql -u {user} -p{passwd} -h {host} -P {port} -N -B -e "{{cmd}}"'
+            db_cmd_fmt = 'mysql -u {user} -p -h {host} -P {port} -N -B -e "{{cmd}}"'
```
Comment (yangdsh) [inline server/website/website/management/commands/resetwebsite.py]: This was written by me previously, and now I found {passwd} useless here.

File: server/website/website/tasks/async_tasks.py [hunk 3]
```diff
@@ -138,7 +138,11 @@ def clean_knob_data(knob_matrix, knob_labels, session):
         for knob in missing_columns:
             knob_object = KnobCatalog.objects.get(dbms=session.dbms, name=knob, tunable=True)
             index = knob_cat.index(knob)
-            matrix = np.insert(matrix, index, knob_object.default, axis=1)
+            try:
+                default_val = float(knob_object.default)
+            except ValueError:
+                default_val = 0
+            matrix = np.insert(matrix, index, default_val, axis=1)
```
Comment (yangdsh) [inline server/website/website/tasks/async_tasks.py]: This is for handling the ValueError caused by converting enum default string to float.

File: server/website/website/views.py [hunk 4]
```diff
@@ -487,20 +487,11 @@ def handle_result_files(session, files):
         knob_data.knobs = JSONUtil.dumps(all_knobs)
 
         data_knobs = JSONUtil.loads(knob_data.data)
+        last_conf = parser.convert_dbms_knobs(result.dbms.pk, last_conf)
```
Comment (yangdsh) [inline server/website/website/views.py]: This handles the error caused by writing enum values when restart fails.

**Comment by yangdsh** [comment]
I don't know how to print the logs of the celery worker. I want to print some knob values but it is not on the screen or in the celery.log. Do you know where to find it or how to enable it? Thanks! @dvanaken

**Comment by bohanjason** [comment]
Have you tried Log.info(knob values) ?  @yangdsh

**Comment by yangdsh** [comment]
> Have you tried Log.info(knob values) ? @yangdsh

Yes. It is not print() but is LOG.info()

File: server/website/website/tasks/async_tasks.py (hunk 3 above)
Comment (bohanjason) [reply server/website/website/tasks/async_tasks.py]: I think it's better to log a error message here if ValueError

Comment (yangdsh) [reply server/website/website/tasks/async_tasks.py]: It is normal for enum & bool type knob to have none min/max. Yet I think we can log the missing column.

**Comment by bohanjason** [comment]
@yangdsh    It looks like the tunable knobs are defined here:
 https://github.com/cmu-db/ottertune/blob/master/server/website/website/set_default_knobs.py

Not sure whether it works for enum or not.

**Comment by dvanaken** [comment]
Thanks @yangdsh @bohanjason !!

**Pull request merged**